### PR TITLE
Update examples to use new API and custom transports

### PR DESCRIPTION
**Pull request by billpittman**
The pub/sub example uses a domain socket transport class to actually send messages between pub->sub.  The transport is fairly primitive and is one way because it wasn't necessary to be two way.

The RPC example uses the SocketUTransport and the python dispatcher, so its a bit more flexible (although more complex because it requires a dispatcher from up-tck).

This is strictly an update of the existing examples, so if you have suggestions to improve it, I'm happy to entertain them.

### CODE REVIEW

**Comment by gregmedd** [comment]
@billpittman - I'd recommend trying this with the [draft socket transport recipe](https://github.com/eclipse-uprotocol/up-conan-recipes/pull/20) to see if the extra files can be eliminated.

File: rpc/src/main_rpc_server.cpp [hunk 1]
```diff
@@ -21,106 +21,89 @@
...
-        return -1;
-    }
+std::optional<uprotocol::datamodel::builder::Payload> OnReceive(
+    const UMessage& message) {
+	// Validate message is an RPC request
+	if (message.attributes().type() != UMessageType::UMESSAGE_TYPE_REQUEST) {
+		spdlog::error("Received message is not a request\n{}",
+		              message.DebugString());
+		return {};
+	}
```
Comment (gregmedd) [inline rpc/src/main_rpc_server.cpp:55]: I _think_ RPC server already does this check

Comment (billpittman) [reply rpc/src/main_rpc_server.cpp:55]: I wasn't sure if it did or not, but seeing as this is an example program, I didn't see the harm in putting in the check.  Maybe I should change the comment to something like:

// Validate message is an RPC request - should not be necessary

**Comment by billpittman** [comment]
@gregmedd I have updated the examples to use the up-transport-socket-cpp package that @agosh01 created, but there was a bug in the recipe, so as of now, this PR is waiting for that PR (https://github.com/eclipse-uprotocol/up-tck/pull/103) to merge.

**Comment by billpittman** [comment]
@gregmedd Nevermind - the PR merged, so all should be fine now.

**Review by gregmedd — APPROVED** [review]
LGTM. We will likely follow up down the line with some streamlining of these examples, but for now they are complete and cover the basic usage of up-cpp.

**Comment by gregmedd** [comment]
Once the `fixup!` commits are squashed, this can be merged.

**Comment by billpittman** [comment]
@gregmedd - This PR is squashed and ready for merging

**Pull request merged**